src/components/molecules/SearchButton.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { FormEvent, ReactElement } from 'react'
+import { ReactComponent as SearchIcon } from '../../images/search.svg'
+import styles from './SearchButton.module.css'
+import { useUserPreferences } from '../../providers/UserPreferences'
+
+export default function SearchButton(): ReactElement {
+  const { isSearchBarVisible, setSearchBarVisible } = useUserPreferences()
+  const isHome = window.location.pathname === '/'
+
+  React.useEffect(() => {
+    if (!isHome) {
+      const searchForm = document?.getElementById('searchForm')
+      if (searchForm) searchForm.focus()
+    }
+  }, [isSearchBarVisible, isHome])
+
+  async function handleButtonClick(e: FormEvent<HTMLButtonElement>) {
+    e.preventDefault()
+    setSearchBarVisible(!isSearchBarVisible)
+    const searchFormSection = document?.getElementById('searchFormSection')
+    if (searchFormSection) {
+      document.getElementById('searchForm').focus()
+      searchFormSection.scrollIntoView()
```

Comment:
Thought (non-blocking): to select specific DOM elements in the "React way" you can take advantage of the ref hook https://it.reactjs.org/docs/hooks-reference.html#useref and then pass it to child components with https://reactjs.org/docs/forwarding-refs.html
In this specific case though, you could probably avoid using any ref at all and simply start the scrolling from within the SearchForm component, reacting to the value change of `isSearchBarVisible` inside a useEffect.

Reply:
Agreed with the suggestion. Since we are using the UserPreference provider already, I would suggest adjusting this to use the `useEffect` hook as suggested by @LucaMilanese90 
Meaning you can drop the `id` in `<SearchForm>` and listen to changes of `isSearchBarVisible` in that component instead. The `useRef` still makes sense to perform the scroll itself even from within the component.